llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -422,39 +431,79 @@ void InterBlockScheduling::enterRegion(MachineBasicBlock *BB,
     BS.addRegion(BB, RegionBegin, RegionEnd);
   }
 }
+
+// Create a block, insert it before Succ, and route the control flow edge
+// between Pred and Succ through it.
+// Since we don't add any control flow instructions, the edge should be a
+// fallthrough edge; it will be replaced with two fallthrough edges and a block
```

Comment:
Where do we verify that the loop falls through to its exit block?

Reply:
In BlockState::classify(), using the layout numbering. If we still have a jump on a fallthrough, we should get rid of that first before thinking about post-pipelining.